app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -59,24 +62,50 @@ class ConfirmReportingMethodSharedController @Inject()(val manageIncomeSources:
                                                        implicit val appConfig: FrontendAppConfig) extends ClientConfirmedController
   with FeatureSwitching with IncomeSourcesUtils {
 
-  def show(id: Option[String],
-           taxYear: String,
+  def show(taxYear: String,
            changeTo: String,
            isAgent: Boolean,
            incomeSourceType: IncomeSourceType
           ): Action[AnyContent] = authenticatedAction(isAgent) { implicit user =>
 
-    handleShowRequest(taxYear, changeTo, isAgent, incomeSourceType, id)
+    withIncomeSourcesFS {
+      sessionService.get(SessionKeys.incomeSourceId).flatMap {
+        case Right(incomeSourceIdMayBe) =>
+          incomeSourceIdMayBe match {
+            case Some(incomeSourceId) => handleShowRequest(taxYear, changeTo, isAgent, incomeSourceType, Some(incomeSourceId))
+            case None => Future.failed(MissingSessionKey(incomeSourceId))
+          }
+        case Left(exception) => Future.failed(exception)
+      }
+    }.recover {
+      case exception =>
+        val errorHandler = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
```

Comment:
you can use method defined in ClientConfirmedController.showInternalServerError in order to reduce number of lines